packages/react-aria-components/stories/Autocomplete.stories.tsx

```diff
@@ -0,0 +1,256 @@
+          <Menu className={styles.menu}>
+            <MyMenuItem onAction={action('Foo action')}>Foo</MyMenuItem>
+            <MyMenuItem onAction={action('Bar action')}>Bar</MyMenuItem>
+            <MyMenuItem onAction={action('Baz action')}>Baz</MyMenuItem>
+          </Menu>
+        </div>
+      </Autocomplete>
+    );
+  },
+  name: 'Autocomplete, onAction on menu items'
+};
+
+const AsyncExample = ({onAction, isDisabled, isReadOnly}) => {
+  let list = useAsyncList<AutocompleteItem>({
+    async load({filterText}) {
+      let json = await new Promise(resolve => {
+        setTimeout(() => {
+          resolve(filterText ? items.filter(item => {
+            let name = item.name.toLowerCase();
+            for (let filterChar of filterText.toLowerCase()) {
+              if (!name.includes(filterChar)) {
+                return false;
+              }
+              name = name.replace(filterChar, '');
+            }
+            return true;
+          }) : items);
+        }, 300);
+      }) as AutocompleteItem[];
+
+      return {
+        items: json
+      };
+    }
+  });
+
+  return (
+    <Autocomplete isDisabled={isDisabled} isReadOnly={isReadOnly} inputValue={list.filterText} onInputChange={list.setFilterText}>
+      <div>
+        <Label style={{display: 'block'}}>Test</Label>
+        <Input />
+        <Text style={{display: 'block'}} slot="description">Please select an option below.</Text>
+        <Menu<AutocompleteItem>
+          items={items}
+          className={styles.menu}
+          onAction={onAction}>
+          {item => <MyMenuItem>{item.name}</MyMenuItem>}
+        </Menu>
+      </div>
+    </Autocomplete>
+  );
+};
+
+export const AutocompleteAsyncLoadingExample = {
+  render: (args) => {
+    return <AsyncExample {...args} />;
+  },
+  name: 'Autocomplete, useAsync level filtering'
+};
+
+
+const CaseSensitiveFilter = ({onAction, isDisabled, isReadOnly}) => {
+  let {contains} = useFilter({
+    sensitivity: 'case'
+  });
+  let defaultFilter = (itemText, input) => contains(itemText, input);
+  return (
+    <Autocomplete isDisabled={isDisabled} isReadOnly={isReadOnly} defaultFilter={defaultFilter}>
+      <div>
+        <Label style={{display: 'block'}}>Test</Label>
+        <Input />
+        <Text style={{display: 'block'}} slot="description">Please select an option below.</Text>
+        <Menu className={styles.menu} items={items} onAction={onAction}>
+          {item => <MyMenuItem id={item.id}>{item.name}</MyMenuItem>}
+        </Menu>
+      </div>
+    </Autocomplete>
+  );
+};
+
+export const AutocompleteCaseSensitive = {
+  render: (args) => {
+    return <CaseSensitiveFilter {...args} />;
+  },
+  name: 'Autocomplete, case sensitive filter'
+};
+
+// TODO: how would we handle this for a DialogTrigger? I guess we could automatically grab the overlaytrigger state from context and send a onAction that gets chained by the one provided
+// to Menu
```

Comment:
Would the onus be on the user to access the Dialog's `close` from the render function and call it in the onAction they provide to the Menu